src/telemetry/telemetry_buffer.ts

```diff
@@ -0,0 +1,79 @@
+// Interfaces
+interface TelemetryEvent  {
+    timestamp : number,
+    action: TypeGenieTelemetryBuffer.Action,
+    value: any
+}
+
+interface TypeGenieTelemetryBufferInterface {
+
+    createTelemetryEvent(timestamp: number, action: TypeGenieTelemetryBuffer.Action, value: any) : TelemetryEvent;
+    addEvent(event: TelemetryEvent) : void;
+    removeEvent(index: number) : void;
+
+}
+
+
+
+export class TypeGenieTelemetryBuffer implements TypeGenieTelemetryBufferInterface{
+
+    private _events : TelemetryEvent[];
+
+    get events() {
+        return this._events;
+    }
+
+    set events(value : TelemetryEvent[]) {
+        this._events = value;
+    }
+
+    constructor() {
+        this.events = [];
+        console.log('Initializing TipeGenieTelemetryBuffer')
+    }
+
+    addEvent(event: TelemetryEvent): void {
+        this._events.push(event)
+    }
+
+    removeEvent(index: number): void {
+        this._events.slice(index,1)
```

Comment:
My understanding is that the `slice` method is not `inplace` operation. As such the original `this._events` will be left unchanged. 

Also, the syntax of `slice(start, end)`, should wouldn't it be something like `slice(index, index+1)` ?